forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,429 @@
+        }
+    };
+
+    Ok(NodeInfoOutput {
+        chain_status,
+        network,
+        uptime,
+        health,
+        wallet_address: default_wallet_address,
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{
+        str::FromStr,
+        sync::Arc,
+        time::{Duration, SystemTime, UNIX_EPOCH},
+    };
+
+    use chrono::Local;
+    use colored::*;
+    use forest_blocks::{BlockHeader, Tipset};
+    use forest_cli_shared::logger::LoggingColor;
+    use forest_shim::{address::Address, econ::TokenAmount};
+    use fvm_shared::clock::EPOCH_DURATION_SECONDS;
+
+    use super::{fmt_info, get_node_status, NodeStatusInfo};
+    use crate::cli::info_cmd::{chain_status, SyncStatus};
+
+    fn mock_tipset_at(duration: u64) -> Arc<Tipset> {
+        let mock_header = BlockHeader::builder()
+            .miner_address(Address::from_str("f2kmbjvz7vagl2z6pfrbjoggrkjofxspp7cqtw2zy").unwrap())
+            .timestamp(duration)
+            .build()
+            .unwrap();
+        let tipset = Tipset::from(&mock_header);
+        Arc::new(tipset)
+    }
+
+    fn node_status_good() -> NodeStatusInfo {
+        super::NodeStatusInfo {
+            behind: 0,
+            health: 90.,
+            epoch: i64::MAX,
+            base_fee: TokenAmount::from_whole(1),
+            sync_status: SyncStatus::Ok,
+        }
+    }
+
+    fn node_status_bad() -> NodeStatusInfo {
+        super::NodeStatusInfo {
+            behind: 0,
+            health: 0.,
+            epoch: 0,
+            base_fee: TokenAmount::from_whole(1),
+            sync_status: SyncStatus::Behind,
+        }
+    }
+
+    #[test]
+    fn node_status_with_null_tipset() {
+        // out of sync
+        let duration_now = SystemTime::now().duration_since(UNIX_EPOCH).unwrap();
+        let tipset = mock_tipset_at(duration_now.as_secs() - 30000);
+        let node_status = get_node_status(&tipset, 0, 0, duration_now).unwrap();
+        assert!(node_status.health.is_nan());
+        assert_eq!(node_status.sync_status, SyncStatus::Behind);
+        // slow
+        let tipset = mock_tipset_at(duration_now.as_secs() - EPOCH_DURATION_SECONDS as u64 * 4);
+        let node_status = get_node_status(&tipset, 10, 1000, duration_now).unwrap();
+        assert!(node_status.health.is_finite());
+        assert_eq!(node_status.sync_status, SyncStatus::Slow);
+        // ok
+        let tipset = mock_tipset_at(duration_now.as_secs() - EPOCH_DURATION_SECONDS as u64);
+        let node_status = get_node_status(&tipset, 10, 1000, duration_now).unwrap();
+        assert!(node_status.health.is_finite());
+        assert_eq!(node_status.sync_status, SyncStatus::Ok);
+    }
+
+    #[test]
+    fn block_sync_timestamp() {
+        let start_time = Local::now();
+        let network = "calibnet";
+        let default_wallet_address = Some("x".to_string());
+        let color = LoggingColor::Never;
+        let duration_now = SystemTime::now().duration_since(UNIX_EPOCH).unwrap();
```

Comment:
You don't need to worry about the system time here.

Reply:
removed